source/agora/consensus/EnrollmentManager.d

```diff
@@ -417,9 +417,17 @@ public class EnrollmentManager
 
     ***************************************************************************/
 
-    public size_t validatorCount () @safe
+    public size_t validatorCount () @safe nothrow
     {
-        return this.validator_set.count();
+        try
+        {
+            return this.validator_set.count();
+        }
+        catch (Exception ex)
+        {
+            log.error("Error while getting validator count: {}", ex.msg);
+            assert(0);
```

Comment:
Just providing a summary of the discussion we had:
Those `assert(0)` are problematic, because they mean "this should **never** happen". Then why is the called function not `nothrow` ?
Usually, this can happen for one main reason: the calling function **can** throw with some parameters, or in some state, but in the state/parameters we're providing, it will never. In this case, there is no parameter provided to `count`, so it has to be tied to the state.
If we look [at the code](https://github.com/bpfkorea/agora/blob/dc869bd16b84188606f3c514b4aefb197e7c26e3/source/agora/consensus/ValidatorSet.d#L120-L139), we see it's just doing a `SELECT`.
So why would it `throw` ? One can make guesses: the lack of type safety means `oneValue` means that it *has* to `throw` (because it's using `std.conv` internally). In this case, perhaps adding a counting function in the sqlite3 bindings would solve the issue. It could also `throw` on non existing tables. Then perhaps there is another construct that would make sure we get a reference to a table that can never be deleted ? Lastly, what if it's simply an IO error ? In this case, all bets are off. If it reads from disk somewhere, and that leads to a `throw`, it **could** happen and thus our `assert(0)` has an invalid assumption.